src/CommunityToolkit.Maui/ImageSources/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,239 @@
+﻿namespace CommunityToolkit.Maui.ImageSources;
+
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Maui.Controls;
+
+/// <summary>Gravatar image source.</summary>
+/// <remarks>Note that <see cref="UriImageSource"/> is sealed and can't be used as a parent!</remarks>
+public class GravatarImageSource : StreamImageSource
+{
+	/// <summary>The backing store for the <see cref="CacheValidity" /> bindable property.</summary>
+	public static readonly BindableProperty CacheValidityProperty = BindableProperty.Create(nameof(CacheValidity), typeof(TimeSpan), typeof(GravatarImageSource), TimeSpan.FromDays(1));
+
+	/// <summary>The backing store for the <see cref="CachingEnabled" /> bindable property.</summary>
+	public static readonly BindableProperty CachingEnabledProperty = BindableProperty.Create(nameof(CachingEnabled), typeof(bool), typeof(GravatarImageSource), true);
+
+	/// <summary>The backing store for the <see cref="Email" /> bindable property.</summary>
+	public static readonly BindableProperty EmailProperty = BindableProperty.Create(nameof(Email), typeof(string), typeof(GravatarImageSource), defaultValue: null, propertyChanged: OnEmailPropertyChanged);
+
+	/// <summary>The backing store for the <see cref="Image" /> bindable property.</summary>
+	public static readonly BindableProperty ImageProperty = BindableProperty.Create(nameof(Image), typeof(DefaultImage), typeof(GravatarImageSource), defaultValue: DefaultImage.MysteryPerson, propertyChanged: OnDefaultImagePropertyChanged);
+
+	/// <summary>The backing store for the <see cref="ParentHeight" /> bindable property.</summary>
+	internal static readonly BindableProperty ParentHeightProperty = BindableProperty.Create(nameof(ParentHeight), typeof(int), typeof(GravatarImageSource), defaultValue: defaultSize, propertyChanged: OnSizePropertyChanged);
+
+	/// <summary>The backing store for the <see cref="ParentWidth" /> bindable property.</summary>
+	internal static readonly BindableProperty ParentWidthProperty = BindableProperty.Create(nameof(ParentWidth), typeof(int), typeof(GravatarImageSource), defaultValue: defaultSize, propertyChanged: OnSizePropertyChanged);
+
+	const int cancellationTokenSourceTimeout = 737;
```

Comment:
it is rather a workaround rather than a solution

Reply:
I discussed this with George before and it comes from wanting to provide a debounce approach to attempt to handle multiple property changes. I think the inspiration came from some .NET MAUI code in `UriImageSource`. Is there a better way to handle this?

Reply:
We may have a flag IsInitialized for each property and do our action depends on it (skip and wait for the next property - when all flags are initialized)

Reply:
I am not sure this delay will work correctly on all devices

Reply:
Not all properties have to be set though

Reply:
We can cache the response for a few seconds which is as for me more reliable than delay

Reply:
The inspiration (which I think was a great suggestion) was from @pictos to use what is already had in place for `UserStoppedTypingBehaviour`. We also make sure that we cancel any existing HTTP requests for completeness.
The delay (with `ContinueWith`) is a compromise for trying to be long enough to catch any property changes and thus prevent multiple HTTP requests, but not so long as to have a noticeable delay effect to the end user.
I'd be happy to look at any suggested alternatives?

Reply:
@VladislavAntonyuk caching the response only helps if nothing changes between requests, the idea of this delay is to only send the latest request and save on HTTP calls. I get that it doesn't feel like the right approach, I went down this route too however through chatting with @GeorgeLeithead we couldn't find a better alternative.

Reply:
Ok, let’s merge it. We’ll find better solution later